torchtune/training/_activation_offloading.py

```diff
@@ -81,6 +81,10 @@ def __init__(
 
         # for streaming
         if use_streams:
+            if torch.__version__ < "2.5.0.dev20240907":
+                raise RuntimeError(
+                    "OffloadActivations with use_streams=True requires PyTorch 2.5.0.dev20240907 or later."
```

Comment:
How do you guys feel about use_treams=True being the default, and if version<2.5, we set it to false and raise a warning (not error)?

Reply:
I think this depends on when this feature will be released in torchtune--if that release will depend on 2.5, then I think defaulting to True makes sense. Otherwise, I think it would not be great to auto-fallback to a slower method when the user may be expecting something else.

Another alternative: we have use_streams default to None, and when unspecified, we check which torch version the running script is using, and we choose True or False depending on that.

Reply:
i think that our stable now will use 2.5

Reply:
> Another alternative: we have use_streams default to None, and when unspecified, we check which torch version the running script is using, and we choose True or False depending on that.

Lol did not see this before making my other comment, but yeah I agree with this.